### PR TITLE
Using avocado diff algorithm instead of 'git fetch'

### PR DESCRIPTION
**Pull request by sergey-shandar**
Currently, `breaking-changes` and `linter-diff` tools do `git fetch` in CI. It causes a problem in AzureDevOps for a private repositories. The fix is replacing `git fetch` with Avocado diff/checkout algorithms.

Example https://github.com/Azure/azure-rest-api-specs/pull/5691

Preproduction Azure pipeline with the changes https://dev.azure.com/azure-sdk/public/_build?definitionId=231

Related Avocado changes: https://github.com/Azure/avocado/pull/18

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -241,14 +219,10 @@ export const getTimeStamp = function() {
...
-  if (prOnly === 'true') {
-    let targetBranch, cmd, filesChanged;
+export const getConfigFilesChangedInPR = async (pr: avocado.PullRequestProperties | undefined) => {
+  if (pr !== undefined) {
     try {
-      targetBranch = getTargetBranch();
-      execSync(`git fetch origin ${targetBranch}`);
-      cmd = `git diff --name-only HEAD $(git merge-base HEAD FETCH_HEAD)`;
-      filesChanged = execSync(cmd, { encoding: 'utf8' }).split('\n');
+      let filesChanged = (await pr.diff()).map(file => file.path)
```
Comment (veronicagg) [inline src/utils.ts]: please add comments in avocado PullRequestUtils on what diff does and what it outputs.

File: src/breaking-change.ts [hunk 2]
```diff
@@ -116,19 +117,21 @@ async function processViaAutoRest(swaggerPath: string) {
 
 //main function
 export async function runScript() {
+  const pr = await avocado.createPullRequestProperties(avocado.defaultConfig())
+
   // See whether script is in Travis CI context
   console.log(`isRunningInTravisCI: ${isRunningInTravisCI}`);
 
   let targetBranch = utils.getTargetBranch();
```
[the remark this answers is not in the document]
Comment (sergey-shandar) [reply src/breaking-change.ts]: we display it somewhere further.

File: src/momentOfTruth.ts [hunk 3]
```diff
@@ -94,39 +89,43 @@ async function getLinterResult(swaggerPath: string|null|undefined) {
...
     console.log(`The results will be logged here: "${logFilepath}".`)
 
-    if (configsToProcess.length > 0) {
+    if (configsToProcess.length > 0 && pr !== undefined) {
         for (const configFile of configsToProcess) {
             await runTools(configFile, 'after');
         }
 
-        await utils.doOnBranch(utils.getTargetBranch(), async () => {
+        await utils.doOnBranch(pr, async () => {
```
[the remark this answers is not in the document]
Comment (sergey-shandar) [reply src/momentOfTruth.ts]: `doOnTargetranch`

**Comment by sergey-shandar** [comment]
@veronicagg https://github.com/Azure/avocado/pull/23/files

**Comment by sergey-shandar** [comment]
@veronicagg it's ready to review again.

File: src/breaking-change.ts [hunk 4]
```diff
@@ -116,19 +117,21 @@ async function processViaAutoRest(swaggerPath: string) {
 
 //main function
 export async function runScript() {
+  const pr = await devOps.createPullRequestProperties(cli.defaultConfig())
+
   // See whether script is in Travis CI context
   console.log(`isRunningInTravisCI: ${isRunningInTravisCI}`);
```
Comment (veronicagg) [inline src/breaking-change.ts]: what would we return if we were running in Travis?

Comment (sergey-shandar) [reply src/breaking-change.ts]: It will not be detected as PR. It's not plan to run in Travis-CI anymore.

**Pull request merged**